projection/proofs/to_cake/payload_to_cakemlProofScript.sml

```diff
@@ -2674,12 +2677,13 @@ val cp_type =
     Just the spec :) *)
 Theorem endpoint_forward_correctness:
   ∀conf p pSt1 pCd1 L pSt2 pCd2
-        vs1 cEnv1 cSt1 vs2 cEnv2 cSt2.
+        vs1 cEnv1 cSt1 cSt2.
     trans conf (NEndpoint p pSt1 pCd1) L (NEndpoint p pSt2 pCd2) ∧
     cpEval_valid conf p cEnv1 pSt1 pCd1 vs1 cSt1 ∧
-    cpEval_valid conf p cEnv2 pSt2 pCd2 vs2 cSt2 ∧
     cpFFI_valid conf pSt1 pSt2 cSt1.ffi.ffi_state cSt2.ffi.ffi_state L ⇒
-    ∃mc. cEval_equiv conf
+    ∃mc vs2 cEnv2.
+         cpEval_valid conf p cEnv2 pSt2 pCd2 vs2 cSt2 ∧
+         cEval_equiv conf
```

Comment:
This allows us  to pick `vs2` and `cEnv2` in the proof which can be quite uninteresting, but note that we still have to provide a suitable `cSt2` which I would argue is desirable since that way we can choose the state we  reach after the transition as long as  its valid under `cpFFI_valid`

Reply:
This makes perfect sense to me.

Reply:
The cSt2.ffi.oracle needs to be constrained; it's otherwise impossible to establish the `cpEval_valid`.